File: Server/src/logic/game/StandardGame.java
package logic.game;

import logic.player.Player;

import java.util.List;

/**
 * Created by Aedan Smith.
 */

public class StandardGame extends Game {
    public StandardGame(List<Player> players){
        super(players);
    }

    @Override
    public void onBegin() {
        for (int i = 0; i < 1; i++) {
            players.forEach(Player::draw);
        }
    }

    @Override
    public void onUpkeep(Player current) {
        current.draw();
    }

    @Override
    public void onEndStep(Player current) {

    }
}

File: Server/src/logic/game/StandardGame.kt
package logic.game

import logic.player.Player
import java.util.function.Consumer

/**
 * Created by Aedan Smith.
 */

class StandardGame(players: MutableList<Player>) : Game(players) {
    override fun onBegin() {
        for (i in 0..0) {
            players.forEach(Consumer<Player> { it.draw() })
        }
    }

    override fun onUpkeep(current: Player) {
        current.draw()
    }

    override fun onEndStep(current: Player) {

    }
}
